test/Dialect/BaseModelica/Transforms/CallCSE/basic-cse.mlir

```diff
@@ -0,0 +1,61 @@
+// RUN: modelica-opt %s --call-cse | FileCheck %s
+
+// CHECK-LABEL: @Test
+module @Test {
+	bmodelica.function @foo {
+		bmodelica.variable @x : !bmodelica.variable<f64, input>
+		bmodelica.variable @y : !bmodelica.variable<f64, output>
+
+		bmodelica.algorithm {
+			%0 = bmodelica.variable_get @x : f64
+			bmodelica.variable_set @y, %0 : f64
+		}
+	}
+
+    // CHECK: bmodelica.model
+	bmodelica.model @M {
+		// CHECK: bmodelica.variable @[[CSE:_cse0_0]]
+		bmodelica.variable @x : !bmodelica.variable<f64>
+		bmodelica.variable @y : !bmodelica.variable<f64>
+
+		%t0 = bmodelica.equation_template inductions = [] {
+			%0 = bmodelica.variable_get @x : f64
+			%lhs = bmodelica.equation_side %0 : tuple<f64>
+			%1 = bmodelica.constant 1.0 : f64
+			// CHECK: %[[RES0:.*]] = bmodelica.variable_get @[[CSE]]
+			// CHECK-NEXT: %[[LHS0:.*]] = bmodelica.equation_side %[[RES0]]
```

Comment:
Not clear why it is named "left-hand side" and then it gets checked for being on the right-hand side.